### PR TITLE
Refactor/generic send request

### PR DESCRIPTION
**Pull request by asdacap**
Refactor as mentioned by the attribute.

## Changes:
- Remove duplicated pattern on send request.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [X] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [X] No

**Comments about testing , should you have some** (optional)

Goerli can sync to head.

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/SyncPeerProtocolHandlerBase.cs [hunk 1]
```diff
@@ -105,35 +104,12 @@ protected virtual async Task<BlockBody[]> SendRequest(GetBlockBodiesMessage mess
...
-                return task.Result;
-            }
-
-            StatsManager.ReportTransferSpeedEvent(Session.Node, TransferSpeedType.Bodies, 0L);
-            throw new TimeoutException($"{Session} Request timeout in {nameof(GetBlockBodiesMessage)} with {message.BlockHashes.Count} block hashes");
+            return await SendRequestGeneric(
+                _bodiesRequests,
+                message,
+                TransferSpeedType.Bodies,
+                static (message) => $"{nameof(GetBlockBodiesMessage)} with {message.BlockHashes.Count} block hashes",
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/SyncPeerProtocolHandlerBase.cs:111]: Is the string computed as a constant on compile?

**Review by LukaszRozmej — APPROVED** [review]
Sidenote we need to drop MessageQueue and use eth66 req/resp mapping

**Comment by asdacap** [comment]
What is that?

**Comment by marcindsobczak** [comment]
> What is that?

https://github.com/NethermindEth/nethermind/issues/4714

**Review by marcindsobczak — APPROVED** [review]

**Pull request merged**